crates/router/src/connector/mifinity/transformers.rs

```diff
@@ -124,109 +278,147 @@ impl<F, T>
+#[serde(rename_all = "camelCase")]
+pub struct MifinityPsyncResponse {
+    payload: Vec<MifinityPsyncPayload>,
 }
 
-impl<F> TryFrom<&MifinityRouterData<&types::RefundsRouterData<F>>> for MifinityRefundRequest {
-    type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(
-        item: &MifinityRouterData<&types::RefundsRouterData<F>>,
-    ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            amount: item.amount.to_owned(),
-        })
-    }
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MifinityPsyncPayload {
+    status: MifinityPaymentStatus,
+    payment_response: Option<PaymentResponse>,
 }
 
-// Type definition for Refund Response
-
-#[allow(dead_code)]
-#[derive(Debug, Serialize, Default, Deserialize, Clone)]
-pub enum RefundStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PaymentResponse {
+    trace_id: Option<String>,
+    client_reference: Option<String>,
+    validation_key: Option<String>,
+    transaction_reference: String,
 }
 
-impl From<RefundStatus> for enums::RefundStatus {
-    fn from(item: RefundStatus) -> Self {
-        match item {
-            RefundStatus::Succeeded => Self::Success,
-            RefundStatus::Failed => Self::Failure,
-            RefundStatus::Processing => Self::Pending,
-            //TODO: Review mapping
-        }
-    }
-}
-
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize)]
-pub struct RefundResponse {
-    id: String,
-    status: RefundStatus,
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum MifinityPaymentStatus {
+    Successful,
+    Pending,
+    Failed,
+    NotCompleted,
 }
 
-impl TryFrom<types::RefundsResponseRouterData<api::Execute, RefundResponse>>
-    for types::RefundsRouterData<api::Execute>
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, MifinityPsyncResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::RefundsResponseRouterData<api::Execute, RefundResponse>,
+        item: types::ResponseRouterData<F, MifinityPsyncResponse, T, types::PaymentsResponseData>,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            response: Ok(types::RefundsResponseData {
-                connector_refund_id: item.response.id.to_string(),
-                refund_status: enums::RefundStatus::from(item.response.status),
+        let status = item
+            .response
+            .payload
+            .first()
+            .map(|payload| payload.to_owned().status.clone())
+            .ok_or(errors::ConnectorError::MissingRequiredField {
+                field_name: "status",
+            })?;
```

Comment:
Should not throw 4xx, this is not what its meant for. 4xx are in case of bad request that user made to our application, we shall only throw it if such scenario happened or connector gave 4xx, you cannot use 4xx to handle your coding scenarios

Reply:
Everywhere this practice is being used and this is wrong, please correct the PR at all the applicable places